src/components/ShadowedScrollView.js

```diff
@@ -0,0 +1,149 @@
+import { StyleSheet, ScrollView, View, Animated } from "react-native";
+import LinearGradient from "react-native-linear-gradient";
+import type { StyleObj } from "react-native/Libraries/StyleSheet/StyleSheetTypes";
+
+type Props = {
+  children: Array<Object>,
+  style: StyleObj
+};
+
+const shadowFadeDuration: number = 100;
+const maxScrollEventThrottle: number = 16;
+const shadowHeight: number = 15;
+
+class ShadowedScrollView extends React.PureComponent<Props> {
+  static defaultProps = {
+    children: [],
+    style: {}
+  };
+
+  // eslint-disable-next-line react/sort-comp
+  topShadowOpacity: Object;
+  bottomShadowOpacity: Object;
+  isTopShadowPresent: boolean = false;
+  isBottomShadowPresent: boolean = true;
+  contentViewHeight: number;
+  scrollViewHeight: number;
+
+  fadeTopShadow = (toValue: number) =>
+    Animated.timing(this.topShadowOpacity, {
+      toValue,
+      duration: shadowFadeDuration,
+      useNativeDriver: true
+    }).start();
+
+  fadeBottomShadow = (toValue: number) =>
+    Animated.timing(this.bottomShadowOpacity, {
+      toValue,
+      duration: shadowFadeDuration,
+      useNativeDriver: true
+    }).start();
+
+  componentWillMount() {
+    this.topShadowOpacity = new Animated.Value(0);
+    this.bottomShadowOpacity = new Animated.Value(1);
+  }
+
+  handleScroll = (event: { nativeEvent: { contentOffset: { y: number } } }) => {
+    const maxScrollOffset = this.contentViewHeight - this.scrollViewHeight - 10;
+    const currentScrollOffset = event.nativeEvent.contentOffset.y;
+
+    if (this.isTopShadowPresent && currentScrollOffset <= 0) {
+      this.isTopShadowPresent = false;
+      this.fadeTopShadow(0);
+    }
+
+    if (!this.isTopShadowPresent && currentScrollOffset > 0) {
+      this.isTopShadowPresent = true;
+      this.fadeTopShadow(1);
+    }
+
+    if (this.isBottomShadowPresent && currentScrollOffset > maxScrollOffset) {
+      this.isBottomShadowPresent = false;
+      this.fadeBottomShadow(0);
+    }
+
+    if (!this.isBottomShadowPresent && currentScrollOffset < maxScrollOffset) {
+      this.isBottomShadowPresent = true;
+      this.fadeBottomShadow(1);
+    }
+  };
+
+  handleScrollViewLayout = (event: Object) => {
+    const { height } = event.nativeEvent.layout;
+    this.scrollViewHeight = height;
+  };
+
+  handleContentViewLayout = (event: Object) => {
+    const { height } = event.nativeEvent.layout;
+    this.contentViewHeight = height;
+  };
+
+  render() {
+    const { style, children } = this.props;
+
+    const topShadowOpacityStyle = {
+      opacity: this.topShadowOpacity.interpolate({
+        inputRange: [0, 1],
+        outputRange: [0, 1]
+      })
+    };
+
+    const bottomShadowOpacityStyle = {
+      opacity: this.bottomShadowOpacity.interpolate({
+        inputRange: [0, 1],
+        outputRange: [0, 1]
+      })
+    };
+
+    return (
+      <View style={style}>
+        <Animated.View style={[styles.topShadow, topShadowOpacityStyle]}>
+          <LinearGradient
```

Comment:
I wonder if you could make the `LinearGradient` itself animated and get rid of the wrapper view by using `Animated.createAnimatedComponent(LinearGradient)`. 🤔

Reply:
It was using that originally, but I remember @joaquimadraz had some issues with it on iOS. I will try to replicate it.

Reply:
Well, that's fine for me. You can leave it like this. Up to you 🙂